roles/srcds/files/addons/sourcemod/scripting/votescramble.sp

```diff
@@ -206,22 +220,25 @@ public int NativeVote_Handler(Handle vote, MenuAction action, int param1, int pa
 	}
 }
 
-// public void Event_RoundEnd(Handle event, const char[] name, bool dontBroadcast) {
-	// if(g_bScrambleTeams) {
-		// float delay = cvarBonusRoundTime.FloatValue - 7.0;
-		// if (delay < 0.0) {
-			// delay = 0.0;
-		// }
-
-		// g_bScrambleTeams = false;
-		// CreateTimer(delay, Timer_Scramble);
-	// }
-// }
-
 public Action Timer_Scramble(Handle timer) {
 	ServerCommand("mp_scrambleteams");
 
-	PrintToChatAll("Scrambling the teams due to vote.");
+	// delay updates of winlimit/maxrounds until next round begins to prevent game from ending immediately during scramble
+	CreateTimer(6.0, Timer_DelayLimitsUpdate);
+
+	MC_PrintToChatAll("[{green}SM{default}] Scrambling the teams due to vote.");
+}
+
+public Action Timer_DelayLimitsUpdate(Handle timer) {
+	// subtract from maxrounds/winlimit after scramble to prevent artificial superextension of maps
+	// assume no limit if 0, don't set negatives
+	if (cvarMaxRounds.IntValue != 0) {
+		SetConVarInt(cvarMaxRounds, cvarMaxRounds.IntValue - g_iRoundsSinceLastScramble, false, true);
+	}
+	if (cvarWinLimit.IntValue != 0) {
+		SetConVarInt(cvarWinLimit, cvarWinLimit.IntValue - (g_iRoundsSinceLastScramble / 2), false, true);
+	}
+	g_iRoundsSinceLastScramble = 0;
```

Comment:
You might want to add

```suggestion
	g_iRoundsSinceLastScramble = 0;
	
	delete timer;
```

here to prevent a memory leak (since you only trigger the timer once and don't keep a reference to it anywhere).

Reply:
done